CIP-XXX/CIP-XXX.md

```diff
@@ -27,10 +27,53 @@ As part of this, we also describe the bit ordering within a `BuiltinByteString`,
 and provide some laws these operations should obey.
 
 ## Motivation: why is this CIP necessary?
+Currently, Plutus Core has a `BuiltinByteString` type, which is a packed array of bytes, similar to Haskell's ByteString.
+We want to implement bit wise operations on `BuiltinByteString` type. In otherwords, we want to achieve bit level granularity on the operations on `BuiltinByteString`.
+Some of the operations that we are interested in are, bit level OR, AND, NOT, or XOR. There are many added advantage of having bit level granularity which we will discuss in more detail later. Unfortunately, all oprations available on the `BuiltinByteString` works on the byte level granularity. So we need this CIP to bridge the gap.
+To explain the usefulness of this granulaity to manipulate `BuiltinByteString`, we use the following example of integer sets.
+#### Integer sets
+Integer sets, also known as bitmap, bitset, or bitvector is used to represent a set of natural numbers. Integer sets can be represented as a `BuiltinByteString` of length $N$, where $N\in \mathbb{N}$ is the maximum possible number in that set. We can represent membership of an element $k \leq N$ by setting the $k^{th}$ bit is set in that `BuiltinByteString`.  For example, assume $ x = \{1,3,4,5\}$ be an integer set. We can represent this set using a byte string of size $8$ as $01011100$. Note that the $1^{st}$, $3^{rd}$, $4^{th}$ and $5^{th}$ bits are set and rest are zero. There are many advantages when we represet a set using Integer set. 
```

Comment:
First sentence lacks plural consistency: you have 'integer set_s_', but then 'bitmap', 'bitvector' and 'bitset' in the singular.

Using a capital-N for length is a bit unusual, both in algorithms literature and in typical writeups of this form: lowercase is more common.

'by setting the kth bit as set'

'For example, let x = ...'.

'A byte string of size 8' is vague: is that 8 bits, or 8 bytes? Also, if we assume that you're representing this as a single byte, your bit order is incorrect (which I believe happens throughout). 

The last sentence needs rephrasing: we can't represent _any_ set this way, 'Integer' shouldn't be capitalized here, and the sentence should end in a colon, not a full stop. 

Lastly, spacing.

Reply:
Regarding the bit order, are we not following 7 down to 0 ? The one you are implementing.